backend/database_handler/transactions_processor.py

```diff
@@ -103,6 +111,11 @@ def insert_transaction(
             s=None,
             v=None,
             leader_only=leader_only,
+            triggered_by=(
+                self.session.query(Transactions).filter_by(hash=triggered_by_hash).one()
+                if triggered_by_hash
+                else None
+            ),
```

Comment:
This query is necessary since SQLAlchemy works with objects, therefore we need to have the object related here